File: 3276-minimum-number-of-pushes-to-type-word-ii/3276-minimum-number-of-pushes-to-type-word-ii.js
/**
 * @param {string} word
 * @return {number}
 */
var minimumPushes = function(word) {
    let freq = new Array(26).fill(0);
    for (let c of word) {
        freq[c.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }

    freq.sort((a, b) => b - a);
    let ans = 0;
    for (let i = 0; i < 26; i++) {
        if (freq[i] === 0) break;
        ans += Math.floor(i / 8 + 1) * freq[i];
    }

    return ans;
};

File: 3276-minimum-number-of-pushes-to-type-word-ii/3276-minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String word) {
        int[] freq = new int[26];
        for (char c : word.toCharArray()) {
            freq[c - 'a']++;
        }

        Arrays.sort(freq);
        int ans = 0;
        for (int i = 25; i >= 0; i--) {
            if (freq[i] == 0) break;
            ans += ((25 - i) / 8 + 1) * freq[i];
        }

        return ans;
    }
}